swig/python/test_comedi_command.py

```diff
@@ -0,0 +1,161 @@
+
+#define SWIGPYTHONONLYSHORT
+
+
+"""
+
+import sys
+import os
+import string
+import argparse
+
+# set the paths so python can find the comedi module
+# Only needed if the C library can't be found, which seems to happen on pytnon3
+if (sys.version_info > (3, 0)):  # python3
+    sys.path.append('/usr/lib/python3/dist-packages/comedi')
+
+import comedi as c
+# This import is later  because the path may need to be appended.
+
+
+class COMEDILIBtest(object):
+    'Test comedilib command functionality'
+
+    def __init__(self, devpath='/dev/comedi0'):
+        # open a comedi device
+        dev = c.open(devpath)
+        if not dev:
+            raise IOError("Error opening Comedi device: %s" % devpath)
+        self.dev = dev
+        # get a file-descriptor for use later
+        self.cfd = c.fileno(dev)
+        self.cmd = None
+
+    def scancommand(self, nscans=1000, chans=None, gains=None, aref=None):
+        """Setup a scan
+        nscans: total number of scans (int)
+        chans: which channels (list)
+        gains: which gain adjustments (list)
+        aref: analog references (list)
+        The lists must all have the same length!
+        """
+        if not chans:
+            raise RuntimeError("No Channels given in 'chans'")
+        if not gains:
+            raise RuntimeError("No gains given in 'gains'")
+        if not aref:
+            raise RuntimeError("No analog references given in 'aref'")
+        if not len(chans) == len(gains) == len(aref):
+            raise RuntimeError(
+                "chans(%d), gains(%d), and aref(%d) not the same size"
+                % (len(chans), len(gains), len(aref)))
+
+        nchans = len(chans)  # number of channels
+
+        # wrappers include a "chanlist" object (just an Unsigned Int
+        # array) for holding the chanlist information
+        mylist = c.chanlist(nchans)  # create a chanlist of length nchans
+
+        # now pack the channel, gain and reference information into
+        # the chanlist object
+        # N.B. the CR_PACK and other comedi macros are now python functions
+        for index in range(nchans):
+            mylist[index] = c.cr_pack(
+                chans[index], gains[index], aref[index])
+
+        # construct a comedi command manually
+        # TODO: make helper methods
+        cmd = c.comedi_cmd_struct()
+        cmd.subdev = 0
+        cmd.flags = 0
+        cmd.start_src = c.TRIG_NOW
+        cmd.sart_arg = 0
+        cmd.scan_begin_src = c.TRIG_TIMER
+        cmd.scan_begin_arg = int(1.0E9 / 100000)
+        cmd.convert_src = c.TRIG_TIMER
+        cmd.convert_arg = 1
+        cmd.scan_end_src = c.TRIG_COUNT
+        cmd.scan_end_arg = nchans
+        cmd.stop_src = c.TRIG_COUNT
+        cmd.stop_arg = nscans
+        cmd.chanlist = mylist
+        cmd.chanlist_len = nchans
+
+        dev = self.dev
+        # test our comedi command a few times.
+        ret = c.comedi_command_test(dev, cmd)
```

Comment:
If you change the file to use the short function names, you should do it for all of them.
`comedi_command_test()` -> `command_test()`
`comedi_cmd_struct()` -> cmd_struct()`
....